internal/memory/backend.go

```diff
@@ -45,14 +48,26 @@ type Store interface {
 func NewBackendStore(
 	ctx context.Context,
 	apiKey, url, ownerName, repoName string,
+	allowHTTP bool,
 	updateInterval time.Duration,
 	serverPort int32,
 ) (Store, error) {
+	client := http.DefaultClient
+	if allowHTTP {
+		client = &http.Client{
+			Transport: &http2.Transport{
+				AllowHTTP: true,
+				DialTLS: func(network, addr string, _ *tls.Config) (net.Conn, error) {
+					return net.Dial(network, addr)
+				},
+			},
+		}
+	}
 	return newBackendStore(
 		ctx,
 		apiKey, ownerName, repoName,
 		updateInterval,
-		backendv1beta1connect.NewDistributionServiceClient(http.DefaultClient, url),
+		backendv1beta1connect.NewDistributionServiceClient(client, url, connect.WithGRPC()),
```

Comment:
does `connect.WithGRPC()` work with `http.DefaultClient`? or do we need to also gate this behind the `allowHTTP` flag?

Reply:
So the default for connect is the connect protocol. Which works fine on our load balancer I think? So wholesale switch to gRPC works just fine.